app/src/main/java/com/github/siela1915/bootcamp/UploadingRecipeFragment.java

```diff
@@ -161,25 +151,27 @@ private void uploadRecipe(final Uri recipeImageUri) {
 //        FirebaseUser user = firebaseAuth.getCurrentUser();
 
         // We are taking the filepath as storagepath + firebaseUser.getUid()+".png"
-        String filepathname = storagepath + "_" + "ZHANG CHI"; // should be firebaseUser.getUid()
-        StorageReference storageReference1 = storageRef.child(filepathname);
-        storageReference1.putFile(recipeImageUri).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
+        String filePathName = storagePath + "_" + "ZHANG CHI"; // should be firebaseUser.getUid()
+        storageRef.child(filePathName).putFile(recipeImageUri).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
             @Override
             public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
                 Task<Uri> uriTask = taskSnapshot.getStorage().getDownloadUrl();
                 while (!uriTask.isSuccessful()) ;
 
-                // We will get the url of our image using uritask
+                // get the url of the recipe image using uritask
                 final Uri downloadUri = uriTask.getResult();
                 if (uriTask.isSuccessful()) {
-                    Map<String, Object> hashMap = getRecipe(downloadUri);
+                    Map<String, Object> hashMap = new HashMap<>();
+                    TextInputLayout recipeName = view.findViewById(R.id.recipeNameContent);
+                    hashMap.put(recipeName.getEditText().getText().toString(), getRecipe(downloadUri));
 
                     // should use firebaseUser.getUid()
                     databaseReference.child("ZHANG CHI").updateChildren(hashMap).addOnSuccessListener(new OnSuccessListener<Void>() {
```

Comment:
Custom branches in the database should be deleted and Database class used instead.